server/controllers/reports_proposed/data/variation_exploitation.js

```diff
@@ -0,0 +1,174 @@
+
+  context.i18n = i18VariationExploitation;
+  context.options = options;
+
+
+  var sql = 
+    'SELECT account.id, account.account_number, account.account_txt, account.parent, account.is_charge, account.account_type_id, (CASE WHEN account.is_charge = 1 THEN SUM(period_total.debit - period_total.credit) ELSE SUM(period_total.credit - period_total.debit) END) AS balance ' +
+    'FROM account LEFT JOIN period_total on account.id = period_total.account_id WHERE account.is_ohada = 1 AND account.account_type_id in (?,?) AND (period_total.fiscal_year_id in (?) OR ISNULL(period_total.fiscal_year_id)) ' +
+    'AND account.id NOT IN (SELECT account.id FROM account WHERE account.account_type_id = ?) GROUP BY account.id';
+
+  db.exec(sql, [1, 3, options.fy, 2])
+  .then(function (ce){ 
+    currentExploitations = ce;
+    return db.exec(sql, [1, 3, options.pfy, 2]);
+  })
+  .then(function (pe){
+    previousExploitations = pe;
+    context.data = process().sort(function (a, b){ return a.account_number - b.account_number;});
+    deferred.resolve(context);
+  })
+  .catch(deferred.reject)
+  .done();
+  return deferred.promise;
+};
+
+function process (){
+  currentSubAccounts = getSubAccountsOnly(currentExploitations);
+  previousSubAccounts = getSubAccountsOnly(previousExploitations);
+
+  currentSubAccounts.forEach(function (item){
+    var parent = getParentAccount(item.parent);
+    var result = getAllChild(parent.id, true);
+    parent.children = result.children;
+    parent.percent = result.percent;
+    parent.balancePrevious = result.balance_previous;
+    parent.viewBalancePrevious = getNumeralFormat(parent.balancePrevious);
+    parent.balanceCurrent = result.balance_curent;
+    parent.viewBalanceCurrent = getNumeralFormat(parent.balanceCurrent);
+    parent.variationValue = getNumeralFormat(parent.balanceCurrent - parent.balancePrevious);
+    finalList.push(parent);
+  });
+
+  previousSubAccounts.forEach(function (item){
+    var parent = getParentAccount(item.parent);
+    var result = getAllChild(parent.id, false);
+    parent.children = result.children;
+    parent.percent = result.percent;
+    parent.balancePrevious = result.balance_previous;
+    parent.viewBalancePrevious = getNumeralFormat(parent.balancePrevious);
+    parent.balanceCurrent = result.balance_curent;
+    parent.viewBalanceCurrent = getNumeralFormat(parent.balanceCurrent);
+    parent.variationValue = getNumeralFormat(parent.balanceCurrent - parent.balancePrevious);
+    finalList.push(parent);
+  });
+
+  return finalList;
+}
+
+function getParentAccount(parentId){
+  var found;
+  for (var i = currentExploitations.length - 1; i >= 0; i--) {
+    if(currentExploitations[i].id == parentId){
+      found = currentExploitations[i];
+      break;
+    }
+  };
+  return found;
+}
+
+function getSubAccountsOnly (list){
+  return list.filter(function (item){
+    return item.account_type_id === 1;
```

Comment:
Is a SubAccount an income/expense account?  Will "income/expense" always have id 1 in every database we install bhima?